client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/contact-form/index.tsx

```diff
@@ -0,0 +1,167 @@
+import { Gridicon } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { Button } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import Form from 'calypso/a8c-for-agencies/components/form';
+import FormField from 'calypso/a8c-for-agencies/components/form/field';
+import QuerySmsCountries from 'calypso/components/data/query-countries/sms';
+import FormPhoneInput from 'calypso/components/forms/form-phone-input';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import { useGetSupportedSMSCountries } from 'calypso/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/contact-editor/hooks';
+import './style.scss';
+
+type FormData = {
+	firstName: string;
+	lastName: string;
+	email: string;
+	agencyName: string;
+	businessUrl: string;
+	phoneNumber: string;
+};
+
+type Props = {
+	onContinue: () => void;
+};
+
+const SignupContactForm = ( { onContinue }: Props ) => {
+	const translate = useTranslate();
+
+	const countriesList = useGetSupportedSMSCountries();
+	const noCountryList = countriesList.length === 0;
+
+	const [ formData, setFormData ] = useState< FormData >( {
+		firstName: '',
+		lastName: '',
+		email: '',
+		agencyName: '',
+		businessUrl: '',
+		phoneNumber: '',
+	} );
+
+	const handlePhoneInputChange = ( data: { phoneNumberFull: string } ) => {
+		setFormData( ( prev ) => ( {
+			...prev,
+			phoneNumber: data.phoneNumberFull,
+		} ) );
+	};
+
+	const handleInputChange =
+		( field: keyof FormData ) => ( event: React.ChangeEvent< HTMLInputElement > ) => {
+			setFormData( ( prev ) => ( {
+				...prev,
+				[ field ]: event.target.value,
+			} ) );
+		};
+
+	const handleSubmit = ( e: React.FormEvent ) => {
+		e.preventDefault();
+		onContinue();
+	};
+
+	return (
+		<Form
+			className="signup-contact-form"
+			title={ translate( 'Sign up' ) }
+			description={ translate(
+				'Join 5000+ agencies and grow your business with Automattic for Agencies. It only takes two minutes!'
+			) }
+		>
+			<div className="signup-multi-step-form__fields">
+				<div className="signup-multi-step-form__name-fields">
+					<FormField label={ translate( 'Your first name' ) } isRequired>
```

Comment:
Can we use `htmlFor` for all the fields on this form? 

We could use `FormLabel` that supports it or add the support on the `FormField` component